dali/pipeline/operators/crop/crop.cu

```diff
@@ -82,81 +79,77 @@ DALIError_t ValidateBatchedCrop(const uint8 * const *in_batch, const int *in_str
   DALI_ASSERT(out_batch != nullptr);
   for (int i = 0; i < N; ++i) {
     DALI_ASSERT(in_batch[i] != nullptr);
-    DALI_ASSERT(in_strides[i] >= C*W);
+    DALI_ASSERT(in_strides[i] >= C * W);
   }
 
   return DALISuccess;
 }
 
-template<>
+template <>
 template <typename Out>
 void Crop<GPUBackend>::RunHelper(Workspace<GPUBackend> *ws, const int idx) {
   const auto output = ws->Output<GPUBackend>(idx);
-  ValidateHelper<Out>(output);
-
-  DALI_CALL((BatchedCrop<Out>(
-      input_ptrs_gpu_.template data<const uint8*>(),
-      input_strides_gpu_.template data<int>(),
-      batch_size_, crop_[0], crop_[1], C_, output_layout_,
-      output->template mutable_data<Out>(),
-      ws->stream())));
+  ValidateHelper<Out>(output, idx);
```

Comment:
I don't know if this is what you mean. For idx is index of argument in input list. So if you call your operator:
op([arg1, arg2]) then idx tells you which arg you are processing now. Here it seems that you thread idx as sample index in the batch.

Reply:
Well spotted. This would actually work for crop.cu  as all indexes contain the same values, however for slice.cu would be troublesome. Accepted